src/lib/queries.js

```diff
@@ -37,18 +37,24 @@ export const observationsQuery = () => ({
   }
 })
 
+// FIXME: Effectively filter messages
 export const recentObservationsQuery = executionId => ({
   definition: () =>
     Q(OBSERVATIONS_DOCTYPE)
-      .indexFields(['updatedAt', 'executionId'])
-      .where({
-        executionId,
-        action: {
-          $ne: 'receiveShare'
-        }
-      })
-      .sortBy([{ updatedAt: 'desc' }])
-      .limitBy(3),
+      // .where({
+      //   executionId: executionId,
+      //   updatedAt: {
+      //     $gt: null
+      //   }
+      // })
+      // .partialIndex({
+      //   action: {
+      //     $ne: 'receiveShare'
+      //   }
+      // })
+      .indexFields(['executionId', 'updatedAt'])
+      .sortBy([{ executionId: 'desc' }, { updatedAt: 'desc' }]),
```

Comment:
Since you do not filter on executionId, I wonder if it would be possible that this ends up in issues regarding the execution, with incorrect observations coming from previous run.
Since there is no more filtering on the 3 last observations, nor on receiveShare, maybe you could try this query? 
```
    Q(OBSERVATIONS_DOCTYPE)
        .where({
          executionId: executionId
        })
      .indexFields(['executionId', 'updatedAt'])
      .sortBy([{ executionId: 'desc' }, { updatedAt: 'desc' }]),
```